tools/automatic_query_fixer/src/test/java/com/google/cloud/bigquery/utils/queryfixer/ParseFactoryTest.java

```diff
@@ -0,0 +1,49 @@
+package com.google.cloud.bigquery.utils.queryfixer;
+
+import org.apache.calcite.sql.SqlKind;
+import org.apache.calcite.sql.SqlNode;
+import org.apache.calcite.sql.parser.SqlParseException;
+import org.apache.calcite.sql.parser.SqlParser;
+import org.apache.calcite.sql.parser.impl.SqlParserImpl;
+
+import com.google.cloud.bigquery.utils.queryfixer.exception.ParserCreationException;
+
+import org.junit.Test;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.fail;
+
+public class ParseFactoryTest {
+
+  @Test
+  public void getParser() throws SqlParseException {
+    BigQueryParserFactory factory = new BigQueryParserFactory();
+    SqlParser parser = factory.getParser("select a from b");
+    SqlNode node = parser.parseQuery();
+    assertEquals(SqlKind.SELECT, node.getKind());
+  }
+
+  @Test
```

Comment:
The best practice for writing tests:

Tests independent behaviors of a method separately:
Example: 
@Test
public void displayTransactionResults_showsItemName() {
  transactionProcessor.displayTransactionResults(
      new User(), new Transaction("Some Item"));
  assertThat(ui.getText()).contains("You bought a Some Item");
}